app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/FriendsActivityTest.java

```diff
@@ -0,0 +1,218 @@
+package com.github.bgabriel998.softwaredevproject;
+
+import android.app.Activity;
+import android.view.View;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.espresso.DataInteraction;
+import androidx.test.espresso.Espresso;
+import androidx.test.espresso.UiController;
+import androidx.test.espresso.ViewAction;
+import androidx.test.espresso.ViewInteraction;
+import androidx.test.espresso.action.ViewActions;
+import androidx.test.espresso.intent.Intents;
+import androidx.test.espresso.intent.matcher.IntentMatchers;
+import androidx.test.espresso.matcher.ViewMatchers;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import com.github.giommok.softwaredevproject.Account;
+import com.github.giommok.softwaredevproject.Database;
+import com.google.firebase.auth.FirebaseAuth;
+
+import org.hamcrest.Matcher;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Locale;
+
+import static androidx.test.espresso.Espresso.onData;
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.swipeDown;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.intent.Intents.intended;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayingAtLeast;
+import static androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertSame;
+
+public class FriendsActivityTest {
+    private final static int FRIENDS_SIZE = 20;
+
+    // Helper method
+    public static ViewAction withCustomConstraints(final ViewAction action, final Matcher<View> constraints) {

```

Comment:
There is a class called something like UITestHelper. Move this function there.

Reply:
Actually, after I modified some code I have not used this method anymore but I forgot to remove it. Thanks for letting me notice it!